lib/semantics/assignment.cc

```diff
@@ -375,23 +374,43 @@ void AssignmentContext::Analyze(const parser::AssignmentStmt &stmt) {
 
 void AssignmentContext::Analyze(const parser::PointerAssignmentStmt &stmt) {
   CHECK(!where_);
-  const auto &lhs{std::get<parser::DataRef>(stmt.t)};
-  const auto &rhs{std::get<parser::Expr>(stmt.t)};
-  auto &foldingContext{context_.foldingContext()};
-  auto lhsExpr{evaluate::Fold(foldingContext, AnalyzeExpr(context_, lhs))};
-  auto rhsExpr{evaluate::Fold(foldingContext, AnalyzeExpr(context_, rhs))};
-  CheckForImpureCall(lhsExpr);
-  CheckForImpureCall(rhsExpr);
-  // TODO: CheckForImpureCall() in the bounds / bounds remappings
-  if (forall_) {
-    // TODO: Warn if some name in forall_->activeNames or its outer
-    // contexts does not appear on LHS
-  }
-  if (lhsExpr && rhsExpr) {
-    CheckForPureContext(*lhsExpr, *rhsExpr, rhs.source, true /* => */);
+  if (const evaluate::Assignment * asst{GetAssignment(stmt)}) {
+    const SomeExpr *lhs;
+    const SomeExpr *rhs;
+    std::visit(
+        common::visitors{
+            [&](const evaluate::Assignment::IntrinsicAssignment &x) {
```

Comment:
These lambdas should return pairs of pointers, I think.  Then there's no question of `lhs` & `rhs` being left uninitialized.